### PR TITLE
Added GPU based wind

### PR DESCRIPTION
**Pull request by Bubbad**
How an object is affected by wind is painted as vertex colors on the object as described in https://mtnphil.wordpress.com/2011/10/18/wind-animations-for-vegetation/ with the difference that the branch attenuation is inverted when coloring, not in shaders.

A scene now has a global wind. Objects are by default not affected by it, but can be if affectedByWind attribute is set on it. 
An object can be affected in three different ways; stem, branch and leaf bending (see article above), which are specified on the object itself. (It needs to be on the GameObject as the renderer sets this information)

### CODE REVIEW

File: includes/IShader.h [hunk 1]
```diff
@@ -30,13 +30,14 @@
 class IShader {
 public:
     virtual void compile() = 0;
-    virtual void checkErrors() = 0;
+    virtual void checkErrors(GLuint &shader) = 0;
```
Comment (Bubbad) [inline includes/IShader.h:33]: Bonus: Fixed a bug where printing shader errors crashed before the message was printed

File: shaders/simple.frag [hunk 2]
```diff
@@ -148,6 +148,10 @@ void main()
     vec3 cubeMapSample = vec3(0.0);
     cubeMapSample = calculateCubeMapSample(directionToEye, normal);
 
+    if(object_alpha < 0.1) {
```
Comment (Bubbad) [inline shaders/simple.frag:151]: Not sure why this appears as diff, it's rebased into the branch from master

File: src/glutil/glutil.cpp [hunk 3]
```diff
@@ -189,7 +189,7 @@ namespace
 #			if defined(_WIN32)
 			__debugbreak();
 #			else // !win32
-			assert(false);
```
Comment (Bubbad) [inline src/glutil/glutil.cpp:192]: This made it crash before printing errors

File: src/objects/Mesh.cpp [hunk 4]
```diff
@@ -119,12 +110,22 @@ void Mesh::initVerticesFromAiMesh(const aiMesh *paiMesh, Chunk &chunk) {
         chunk.m_normals.push_back(make_vector(pNormal.x, pNormal.y, pNormal.z));
         chunk.m_uvs.push_back(make_vector(pTexCoord.x, pTexCoord.y));
 
-        if(paiMesh->HasVertexColors(0)) {
-            aiColor4D vertexColor = paiMesh->mColors[0][i];
-            chunk.m_vertexColors.push_back(chag::make_vector(vertexColor.r, vertexColor.g, vertexColor.b));
-        } else {
-            chunk.m_vertexColors.push_back(chag::make_vector(0.0f, 0.0f, 0.0f));
-        }
+        unsigned int j = 0;
```
Comment (Bubbad) [inline src/objects/Mesh.cpp]: Previously, we only loaded the first vertex color buffer, but there can be many!

File: src/core/Renderer.cpp [hunk 5]
```diff
@@ -207,21 +214,31 @@ void Renderer::setLights(std::shared_ptr<ShaderProgram> &shaderProgram, Scene *s
...
         glDepthFunc(GL_LESS);
-        shaderProgram->setUniform1f("object_reflectiveness", (*transparentObjects[i]).shininess);
-        drawModel(*transparentObjects[i], shaderProgram);
+        glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+        shaderProgram->setUniform1f("object_reflectiveness", transparentObject->shininess);
+        shaderProgram->setUniform1i("isAffectedByWind", transparentObject->getIsAffectedByWind());
+        shaderProgram->setUniform1f("branchAmplitude", transparentObject->getBranchAmplitude());
+        shaderProgram->setUniform1f("leafAmplitude", transparentObject->getLeafAmplitude());
+        shaderProgram->setUniform1f("mainBendiness", transparentObject->getMainBendiness());
+        drawModel(transparentObject, shaderProgram);
```
Comment (DunderRoffe) [inline src/core/Renderer.cpp]: code duplication

File: src/glutil/glutil.cpp [hunk 6]
```diff
@@ -189,7 +189,7 @@ namespace
...
+//			assert(false);
```
Comment (DunderRoffe) [inline src/glutil/glutil.cpp]: Remove code instead of commenting it out.

File: src/objects/Mesh.cpp [hunk 7]
```diff
@@ -119,12 +110,22 @@ void Mesh::initVerticesFromAiMesh(const aiMesh *paiMesh, Chunk &chunk) {
...
+        do {
+            if (paiMesh->HasVertexColors(j)) {
```
Comment (DunderRoffe) [inline src/objects/Mesh.cpp]: The contents of this do loop should be extracted to a function for read ability

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

**Review by DunderRoffe — APPROVED** [review]

**Pull request merged**